parse_lib.py

```diff
@@ -95,8 +94,9 @@ def resolve_relative_resource_urls(html_string):
     imgs = html.find_all("img", src=True)
     equations = html.find_all("object", data=True)
     list(map(update_anchor_href, anchors))
```

Comment:
Awesome! Maybe do a search across the whole project---also, I think I saw a few list-comprehensions a while back that had side-effects.  We should split these out too while we are at it.